src/library/scala/collection/immutable/StrictOptimizedSeqOps.scala

```diff
@@ -27,7 +27,7 @@ trait StrictOptimizedSeqOps[+A, +CC[_], +C]
   }
 
   override def updated[B >: A](index: Int, elem: B): CC[B] = {
-    if (index < 0) throw new IndexOutOfBoundsException(index.toString)
+    if (index < 0) throw new IndexOutOfBoundsException(if (knownSize >= 0) s"can't update at $index since bound is 0-${knownSize - 1}" else index.toString)
```

Comment:
Since we know the problem -- index can't be negative -- we can say that directly